samples/snippets/src/main/java/pubsub/CreateTopicWithSchemaExample.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_create_topic_with_schema]
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.pubsub.v1.Encoding;
+import com.google.pubsub.v1.SchemaName;
+import com.google.pubsub.v1.SchemaSettings;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+
+public class CreateTopicWithSchemaExample {
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String topicId = "your-topic-id";
+    // Use an existing schema.
+    String schemaId = "your-schema-id";
+    // Choose either BINARY or JSON encoding.
+    Encoding encoding = Encoding.BINARY;
```

Comment:
Why is the encoding here `BINARY`, when the schema definition in `CreateAvroSchemaExample` is `JSON`?

Reply:
This encoding applies to message serialization. Earlier on, the type of the Avro schema file is JSON. An Avro schema file is always represented in JSON.

Reply:
Ah, I see now, that was clear from the API docs. Sorry, thank you.

> The encoding of messages validated against `schema`.